snooty/test_parser.py

```diff
@@ -615,3 +615,97 @@ def test_cond() -> None:
         </directive>
         </root>""",
     )
+
+
+def test_version() -> None:
+    path = ROOT_PATH.joinpath(Path("test.rst"))
+    project_config = ProjectConfig(ROOT_PATH, "", source="./")
+    parser = rstparser.Parser(project_config, JSONVisitor)
+
+    page, diagnostics = parse_rst(
+        parser,
+        path,
+        """
+.. versionchanged:: 2.0
+    This is some modified content.
+
+    This is a child that should also be rendered here.
+
+This paragraph is not a child.
+""",
+    )
+    page.finish(diagnostics)
+    assert len(diagnostics) == 0
+    check_ast_testing_string(
+        page.ast,
+        """<root>
+        <directive name="versionchanged">
+        <text>
+        2.0
+        </text>
+        <text>
+        This is some modified content.
+        </text>
+        <paragraph>
+        <text>This is a child that should also be rendered here.</text>
+        </paragraph>
+        </directive>
+        <paragraph>
+        <text>This paragraph is not a child.</text>
+        </paragraph>
+        </root>""",
+    )
+
+    page, diagnostics = parse_rst(
+        parser,
+        path,
+        """
+.. deprecated:: 1.8
+
+A new paragraph.
+""",
+    )
+    page.finish(diagnostics)
+    assert len(diagnostics) == 0
+    check_ast_testing_string(
+        page.ast,
+        """<root>
+        <directive name="deprecated">
+        <text>
+        1.8
+        </text>
+        </directive>
+        <paragraph>
+        <text>A new paragraph.</text>
+        </paragraph>
+        </root>""",
+    )
+
+    page, diagnostics = parse_rst(
```

Comment:
What's the difference between this and the first test?

Reply:
I wanted to have tests for all directives: `versionadded`, `versionchanged`, and `deprecated`. But perhaps it is redundant?

Reply:
It's all the same implementation, so it's really not necessary. But I don't think it's worth doing another round for :+1: